platform/common/export/common/gpio.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef COMMON_GPIO_HPP_
+#define COMMON_GPIO_HPP_
+
```

Comment:
You right. 
I first thought that pins ports and numbers will be shared across platforms of even used outside of platform.

But we have `gpio` class, that already encapsulates GPIO usage under common interface. So common/gpio.hpp can be dropped.